libebpfdiscoveryskel/src/Handlers.h

```diff
@@ -143,16 +143,18 @@ __attribute__((always_inline)) inline static void handleRead(
 		struct pt_regs* ctx,
 		struct DiscoveryGlobalState* globalStatePtr,
 		struct DiscoveryAllSessionState* allSessionStatePtr,
-		struct ReadArgs* readArgsPtr,
-		ssize_t bytesCount) {
+		char* buf,
+		__u32 fd,
+		ssize_t bytesCount,
+		__u64 pidTgid) {
```

Comment:
This function has to much params, please pass it via struct. Perhaps `ReadArgs` could be reused?

Reply:
`ReadArgs` should ad least contain `bytesCount` - currently we have buffer there but not buffer size...

Reply:
We used to pass `ReadBufArgs` here previously, but it only kept-in two params. Removing it cleared out some structure initialization logic outside of the handler. The change was made [here](https://github.com/dynatrace-oss/eBPF-Discovery/pull/77/commits/58d1d59d221f7710337e01a4b7a3cda15af1ad97).

Reply:
I have added `ReadSyscallHandlerArgs` to pass read syscall args to a handler.
There are also `ReadSyscallScalarArgs` (used to pass read syscall args with one buffor) and `ReadSyscallVectorArgs` (used to pass read syscall args with multiple buffors in a vector structure).